frontend/src/lib/utils.tsx

```diff
@@ -29,6 +29,17 @@ export function uuid(): string {
     )
 }
 
+export function isObjectEmpty(obj: Record<string, any>): boolean {
+    return obj && Object.keys(obj).length === 0 && obj.constructor === Object
+}
+
+// Builds url with base + search params + hash params
+export function buildUrl(base: string, searchParams: Record<string, any>, hashParams: Record<string, any>): string {
+    const search = !isObjectEmpty(searchParams) ? `?${toParams(searchParams)}` : ''
+    const hash = !isObjectEmpty(hashParams) ? `#${toParams(hashParams)}` : ''
+    return `${base.replace(/\/$/, '')}${search}${hash}`
+}
+
```

Comment:
It's better to reuse the `combineUrl` function exported by `kea-router` for this.

Reply:
Oo nice, I'll be damned if there's something kea can't do 😆